src/bridge/cockpitpipechannel.c

```diff
@@ -130,6 +153,27 @@ cockpit_pipe_channel_control (CockpitChannel *channel,
           goto out;
         }
 
+      /* ignore size options if this channel is not a pty or we are in prepare() */
+      if (self->pty && self->pipe)
+        {
+          gushort rows, cols;
+
+          if (!cockpit_pipe_channel_read_window_size_options (message, 0, 0, &rows, &cols))
+            {
+              g_warning ("invalid \"window_rows\" or \"window_cols\" option for stream channel");
+              goto out;
+            }
+
+          if (rows > 0 && cols > 0)
+            {
+              struct winsize size = { rows, cols, 0, 0 };
+              gint fd;
+
+              g_object_get (self->pipe, "in-fd", &fd, NULL);
+              ioctl (fd, TIOCSWINSZ, &size);
```

Comment:
We should print out with a g_warning() here (if we expect this never to fail on a valid system).